File: lab/ki305/kohut/lab8/writer.py
import os
import struct

class Writer:
    def __init__(self, result=0.0):
        """Ініціалізує Writer з початковим результатом."""
        self.result = result

    def set_result(self, result):
        """Метод для встановлення значення результату."""
        self.result = result

    def get_result(self):
        """Метод для отримання значення результату."""
        return self.result

    def wr_txt(self, f_name):
        """Запис результату у текстовий файл."""
        with open(f_name, 'w') as f:
            f.write(f"{self.result:.6f}\n")

    def rd_txt(self, f_name):
        """Зчитування результату з текстового файлу."""
        if os.path.exists(f_name):
            with open(f_name, 'r') as f:
                self.result = float(f.readline().strip())
            return self.result
        raise FileNotFoundError(f"Файл {f_name} не знайдено")

    def wr_bin(self, f_name):
        """Запис результату у двійковий файл."""
        with open(f_name, 'wb') as f:
            f.write(struct.pack('d', self.result))

    def rd_bin(self, f_name):
        """Зчитування результату з двійкового файлу."""
        if os.path.exists(f_name):
            with open(f_name, 'rb') as f:
                self.result = struct.unpack('d', f.read(8))[0]
            return self.result
        raise FileNotFoundError(f"Файл {f_name} не знайдено")

File: lab/ki305/kohut/lab5/Writer.java
package ki305.kohut.lab5;

import java.io.*;

/**
 * Клас Writer - відповідає за запис та читання результатів у текстовий і двійковий файли.
 */
public class Writer {
    private double result;

    // Конструктор за замовчуванням
    public Writer() { }

    // Конструктор з параметром результату
    public Writer(double result) {
        this.result = result;
    }

    // Метод для встановлення значення результату
    public void setResult(double result) {
        this.result = result;
    }

    // Метод для отримання значення результату
    public double getResult() {
        return this.result;
    }

    // Метод для запису результату у текстовий файл
    public void wrTxt(String fName) throws IOException {
        try (PrintWriter writer = new PrintWriter(fName)) {
            writer.printf("%f", result); // Запис результату у текстовий файл
        }
    }

    // Метод для зчитування результату з текстового файлу
    public double rdTxt(String fName) throws IOException {
        try (BufferedReader reader = new BufferedReader(new FileReader(fName))) {
            result = Double.parseDouble(reader.readLine()); // Читання результату
        }
        return result;
    }

    // Метод для запису результату у двійковий файл
    public void wrBin(String fName) throws IOException {
        try (DataOutputStream dos = new DataOutputStream(new FileOutputStream(fName))) {
            dos.writeDouble(result); // Запис результату у двійковий файл
        }
    }

    // Метод для зчитування результату з двійкового файлу
    public double rdBin(String fName) throws IOException {
        try (DataInputStream dis = new DataInputStream(new FileInputStream(fName))) {
            result = dis.readDouble(); // Читання результату
        }
        return result;
    }
}
